### PR TITLE
[String] Add SpanishInflector support for singular and plural

### PR DESCRIPTION
**Pull request by dennistobar**
This PR implement a new Inflector using Spanish (ISO Code = es) implemented in this PR symfony/symfony#58228.

* I sort the list of supported Inflectors to follow some rule (alphabet?)
* I just change invalid test from "it" to "qq" (qq doesn't exist as valid ISO code, but it is italian... don't be evil if anyone want to create the italian inflector 😅 )

### CODE REVIEW

File: extra/string-extra/StringExtension.php [hunk 1]
```diff
@@ -79,10 +81,12 @@ public function singular(string $value, string $locale = 'en', bool $all = false
     private function getInflector(string $locale): InflectorInterface
     {
         switch ($locale) {
-            case 'fr':
-                return $this->frenchInflector ?? $this->frenchInflector = new FrenchInflector();
             case 'en':
                 return $this->englishInflector ?? $this->englishInflector = new EnglishInflector();
+            case 'es':
+                return $this->spanishInflector ?? $this->spanishInflector = new SpanishInflector();
```
Comment (stof) [inline extra/string-extra/StringExtension.php:91]: this needs to handle the case where SpanishInflector does not exist in `symfony/string`, as it does not exist for all versions supported by the requirement.

**Review by stof — CHANGES_REQUESTED** [review]

File: extra/string-extra/StringExtension.php (hunk 1 above)
Comment (dennistobar) [reply extra/string-extra/StringExtension.php:91]: Hi, thanks for the comment, do you any example to make the check?, and how can I test it?.

Comment (stof) [reply extra/string-extra/StringExtension.php:91]:
```php
if (!class_exists(SpanishInflector::class)) {
     // throw exception with an helpful message
}
```

For testing, you would have to define 2 different tests for the 2 cases, with different `CONDITION` sections to skip the one that does not correspond to the availability of the class

File: extra/string-extra/Tests/Fixtures/singular-invalid-language.test [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
 "singular" filter
 --TEMPLATE--
-{{ 'partitions'|singular('it') }}
+{{ 'partitions'|singular('qq') }}
 
 --DATA--
 return []
 
 --EXCEPTION--
-Twig\Error\RuntimeError: An exception has been thrown during the rendering of a template ("Locale "it" is not supported.") in "index.twig" at line 2.
```
Comment (fabpot) [inline extra/string-extra/Tests/Fixtures/singular-invalid-language.test]: Can you revert these changes were you've removed the end line?

Comment (dennistobar) [reply extra/string-extra/Tests/Fixtures/singular-invalid-language.test]: done

**Comment by fabpot** [comment]
Thank you @dennistobar.
